src/middlewared/middlewared/service.py

```diff
@@ -871,6 +872,32 @@ def ping(self):
         """
         return 'pong'
 
+    def _ping_host(self, host, count, timeout):
+        if osc.IS_LINUX:
+            process = run(['ping', '-c', f'{count}', '-w', f'{timeout}', host])
+        else:
+            process = run(['ping', '-c', f'{count}', '-t', f'{timeout}', host])
+
+        if process.returncode != 0:
+            return False
+        else:
+            return True
+
+    @accepts(
+        Str('hostname'),
+        Int('timeout', default=10),  # seconds
+    )
+    def ping_remote(self, hostname, timeout):
+        """
```

Comment:
Let's fix up the docstring.

`Method that will send a ICMPv4 echo request to "hostname" and will wait up to "timeout".`